source/iface/sampled-image-accessor.rst

```diff
@@ -2,20 +2,220 @@
+``sampled_image_accessor`` can be used: ``image_target::device``
+means the accessor can be used in a SYCL kernel function while
+``image_target::host_task`` means the accessor can be used in
+a host task. Programs which specify this template parameter as
+``image_target::device`` and then use the ``sampled_image_accessor``
+from a host task are ill formed. Likewise, programs which specify this
+template parameter as ``image_target::host_task`` and then use the
+``sampled_image_accessor`` from a SYCL kernel function are ill formed.
+
 .. _sampled_image_accessor:
 
-==================================
+================================
 ``sycl::sampled_image_accessor``
-==================================
+================================
+
+::
+
+  template <typename DataT, int Dimensions,
+            image_target AccessTarget = image_target::device>
+  class sampled_image_accessor;
+
+(constructor of the `sampled_image_accessor`)
+=============================================
+
+.. parsed-literal::
+
+  template <typename AllocatorT>
+  sampled_image_accessor(sampled_image<Dimensions, AllocatorT>& imageRef,
+                         handler& commandGroupHandlerRef,
+                         const property_list& propList = {})
+
+Constructs a ``sampled_image_accessor`` for accessing a
+``sampled_image`` within a command on the ``queue``
+associated with ``commandGroupHandlerRef``. The optional
+``property_list`` provides properties
+for the constructed object.
+
+If ``AccessTarget`` is ``image_target::device``,
+throws an ``exception`` with the ``errc::feature_not_supported``
+error code if the device associated with ``commandGroupHandlerRef``
+does not have ``aspect::image``.
 
 .. _host_sampled_image_accessor:
 
 =======================================
 ``sycl::host_sampled_image_accessor``
 =======================================
+
+::
+
+  template <typename DataT, int Dimensions> class host_sampled_image_accessor;
+
+(constructor of the `host_sampled_image_accessor`)
+====================================================
+
+.. parsed-literal::
+
+  template <typename AllocatorT>
+  host_sampled_image_accessor(sampled_image<Dimensions, AllocatorT>& imageRef,
+                              const property_list& propList = {})
+
+Constructs a ``host_sampled_image_accessor`` for accessing an
+``sampled_image`` immediately on the host. The optional
+``property_list`` provides properties for the constructed object.
+
+=============================================
+Member functions of the sampled image classes
+=============================================
+
+Both classes have member functions with the same name,
+which are described below.
+
+``size``
+========
+
+::
+
+  size_t size() const noexcept
+
+Returns the number of elements of the underlying
+``sampled_image`` that this accessor is accessing.
+
+``read``
+========
+
+::
+
+  template <typename CoordT> DataT read(const CoordT& coords) const
```

Comment:
```suggestion
  template <typename CoordT> DataT
  read(const CoordT& coords) const
```

For this reference, it's probably better to split lines like suggested here, for readability.

Reply:
Fixed with https://github.com/KhronosGroup/SYCL_Reference/pull/106/commits/0fd5efe01092231cbb851420c283210d0f1c1cb0